soundbay/utils/logging.py

```diff
@@ -175,12 +182,52 @@ def upload_artifacts(self, audio: torch.Tensor, label: torch.Tensor, raw_wav: to
     @staticmethod
     def get_metrics_dict(label_list, pred_list, pred_proba_array):
         """calculate the metrics comparing the predictions to the ground-truth labels, and return them in dict format"""
-        accuracy = metrics.accuracy_score(label_list, pred_list)  # calculate accuracy using sklearn.metrics
-        f1score = metrics.f1_score(label_list, pred_list)
-        precision = metrics.precision_score(label_list, pred_list)
-        recall = metrics.recall_score(label_list, pred_list)
-        auc = metrics.roc_auc_score(label_list, pred_proba_array[:, 1])
-        metrics_dict = {'accuracy': accuracy, 'f1score': f1score, 'precision': precision, 'recall': recall, 'auc': auc}
+
+        if isinstance(label_list, list):
+            label_list = np.array(label_list)
```

Comment:
I think it's better to add types signatures to the method definition and convert lists to np.arrays always.
The function should have defined api, I prefer not to have `Union[list, np.ndarray]` unless we have a really good reason for that